pkg/storage/s3.go

```diff
@@ -0,0 +1,84 @@
+package storage
+
+import (
+	"bytes"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"golang.org/x/net/context"
+)
+
+const ()
+
+// newS3Backend returns a new backend implementation backed by the supplied
+// S3 bucket name. Note that with the current interface, the AWS_REGION environment
+// variable must be specified to use this backend.
+func newS3Backend(bucketName string) (*s3Backend, error) {
+	return &s3Backend{bucketName: aws.String(bucketName)}, nil
+}
+
+type s3Backend struct {
+	bucketName *string
+}
+
+func (b *s3Backend) read(_ context.Context, id string) ([]byte, error) {
+	sess := session.Must(session.NewSession())
+
+	s3Client := s3.New(sess)
+	getOutput, err := s3Client.GetObject(&s3.GetObjectInput{
```

Comment:
Seems like there's an easily accessible `GetObjectWithContext` method [1]. Given that we already have a context passed in, could we call that instead? Here and below.

[1] https://github.com/aws/aws-sdk-go/blob/master/service/s3/api.go#L2969

Reply:
It's pretty great that they support this now 🎉